app/src/pages/eligibility.tsx

```diff
@@ -42,33 +36,54 @@ const Eligibility: NextPage<Props> = (props: Props) => {
     /* NOTE: We are using useEffect() because we want to make sure the props provided by getServerSideProps() are reliably loaded into the page. */
     const prevRouteIndex = props.previousRoute.lastIndexOf('/')
     const previousRoute = props.previousRoute.substring(prevRouteIndex)
-    if (form.none) {
+    if (form.categorical.none) {
       setContinueBtn({ ...continueBtn, route: '/other-benefits' })
     } else if (previousRoute === '/review') {
       setContinueBtn({
         labelKey: 'updateAndReturn',
         route: previousRoute,
       })
     } else setContinueBtn({ ...continueBtn, route: incomeRoute })
-  }, [form.none, props.previousRoute])
+  }, [form.categorical.none, props.previousRoute])
 
   useEffect(() => {
     setDisabled(!requiredMet())
-  }, [form])
+  }, [requiredMet(), form])
 
   const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
     const { value, name }: { value: string; name: string } = e.target
-    const castValue = value as keyof typeof form
-    let newValue
+    let newForm: typeof form = form
+    let newValue = {}
 
     if (['residential', 'before'].includes(name)) {
-      newValue = { [name]: value }
+      set(newValue, name, value)
     } else {
-      newValue = { [castValue]: !form[castValue] }
+      const castValue = value as keyof typeof form
+      const categoricalOrProgram: 'categorical' | 'programs' = castValue.split(
+        '.'
+      )[0] as 'categorical' | 'programs'
+      const nestedNoneKey = `${categoricalOrProgram}.none`
+      newValue = { [categoricalOrProgram]: form[categoricalOrProgram] }
+      // toggles deeply nested boolean of categorical or program value
+      set(newValue, castValue, !getNestedValue(form, castValue))
+      if (
+        castValue.includes('none') &&
+        getNestedValue(newForm, nestedNoneKey)
+      ) {
+        // If none was just set to true, deselect all other options
+        Object.keys(form[categoricalOrProgram]).forEach((key) => {
+          if (key !== 'none') {
+            const nestedKey = `${categoricalOrProgram}.${key}`
+            set(newForm, nestedKey, false)
+          }
+        })
+      } else if (getNestedValue(form, nestedNoneKey)) {
+        // if none is already true for current question, deselect it
+        set(newForm, nestedNoneKey, false)
+      }
```

Comment:
I think you're on the right track here, but the way you're handling the `value` feels brittle to me. I think it's better to modify the data structure to be able to better handle the data without needing to meta program around the `.` splitting. In [my draft PR](https://github.com/navapbc/wic-mt-demo-project-eligibility-screener/pull/136/files#diff-5edb4bd7dec829578b7b6c665139563e289841a6453f22d3646afe797e2ee8ee), I've refactored the categorical eligibility criteria and the adjunctive eligibility criteria into arrays, which would let us modify the data more easily.

Reply:
@rocketnova should I keep this PR open until your refactor is merged and I can update accordingly?